service/service.hpp

```diff
@@ -185,21 +188,44 @@ namespace raven
         DLOG_IF_F(INFO, cfg.config_read_only_key.has_value(), "cfg.config_read_only_key: %s",
                   cfg.config_read_only_key.value().value().c_str());
 
-        auto answer = db_.config_load(cfg);
-        if (answer.request_state == convert_request_state.at(request_state::success)) {
-            config_clients_registry_.at(sock.fileno()) += answer.config_id;
-            answer.config_id = config_clients_registry_.at(sock.fileno()).get_last_id();
+        config_id_st id;
+        if (cfg.config_key) {
+            id = db_.get_config_id(cfg.config_key.value());
+        } else if (cfg.config_read_only_key) {
+            id = db_.get_config_id(cfg.config_read_only_key.value());
+        } else {
+            send_answer(sock, request_state::unknown_request);
+            return ;
         }
-        prepare_answer(sock, answer);
+        if (db_.fail()) {
+            if (db_.get_state() == db_state::unknow_config_key) {
+                send_answer(sock, request_state::unknown_key);
+                return ;
+            }
```

Comment:
I'd use an else here and put the return below the if else. I don't like the idea of an early return in an if when it's scope is so small.

Reply:
For now only one particular db state require to send a particular request state, where all other reason of failing send back a db_error, but in the future we might have more than one special case. so I thought checking them with ifs with early returns and a default case, maybe a switch will be better ( I thought it wasn't necessary because there was only one special case but now I think its a bit better)

Reply:
I like the idea of a switch. Makes the intent clearer.

Reply:
done